File: 229-majority-element-ii/majority-element-ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n=len(nums)
        t=int(n/3)
        ans=[]
        my={}
        for i in nums:
            my[i]=my.get(i,0)+1
        for j,k in my.items():
            if k > t:
                ans.append(j)
        return ans
        

File: 229-majority-element-ii/majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
         int n = nums.size();
        int t = n / 3;
        std::vector<int> ans;
        std::unordered_map<int, int> my;

        for (int i : nums) {
            my[i]++;
        }

        for (auto& pair : my) {
            if (pair.second > t) {
                ans.push_back(pair.first);
            }
        }

        return ans;
    }
};